src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -118,6 +120,16 @@ impl RowSetIterator {
             }
             min.unwrap_or(ROWSET_MAX_OUTPUT)
         };
+        if self.only_scan_row_handler {
+            fetch_size = self
+                .rowset
+                .column(0)
+                .index()
+                .indexes()
+                .iter()
+                .map(|x| x.row_count)
+                .sum::<u32>() as usize;
```

Comment:
summing the total row doesn't look like a trivial operation. There could be thousands of blocks in a single column. I think we can use the original `fetch_size` for row handlers. Most executors prefer to consume chunks little by little, instead of getting them all at once.

Also, the `expected_size` of `next_batch` function requires that `next_batch` cannot return elements more than `expected_size`. This seems to violate the rule.